libs/SalesforceReact/src/com/salesforce/androidsdk/reactnative/ui/SalesforceReactActivity.java

```diff
@@ -102,50 +96,80 @@ public void onPause() {
         passcodeManager.onPause(this);
     }
 
-    @Override
-    public void onDestroy() {
-        unregisterReceiver(userSwitchReceiver);
-        super.onDestroy();
+    public void logout(Callback successCallback) {
+        Log.i(TAG, "logout called");
+        SalesforceSDKManager.getInstance().logout(this);
+        if (successCallback != null) {
+            successCallback.invoke();
+        }
     }
 
-    /**
-     * Refreshes the client if the user has been switched.
-     */
-    protected void refreshIfUserSwitched() {
-        if (passcodeManager.onResume(this)) {
+    public void authenticate(final Callback successCallback, final Callback errorCallback) {
+        Log.i(TAG, "authenticate called");
+         clientManager.getRestClient(this, new RestClientCallback() {
+
+             @Override
+             public void authenticatedRestClient(RestClient client) {
+                 if (client == null) {
+                     Log.i(TAG, "authenticate - authenticatedRestClient called with null client");
+                     logout(null);
+                 } else {
+                     Log.i(TAG, "authenticate - authenticatedRestClient called with actual client");
+                     SalesforceReactActivity.this.client = client;
+
+
+	                /*
+                     * Do a cheap REST call to refresh the access token if needed.
+                     * If the login took place a while back (e.g. the already logged
+                     * in application was restarted), then the returned session ID
+                     * (access token) might be stale.
+                     */
+                     client.sendAsync(RestRequest.getRequestForResources(ApiVersionStrings.getVersionNumber(SalesforceReactActivity.this)), new RestClient.AsyncRequestCallback() {
```

Comment:
We do the same for cordova.
If you use the session token directly, it better be fresh.
That code could also be moved to the getAuthCredentials method.